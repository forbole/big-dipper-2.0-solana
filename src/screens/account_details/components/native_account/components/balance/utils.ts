import { formatNumber } from '@utils/format_token';
import { BalanceType } from './types';

export const formatBalanceData = (data: BalanceType) => {
  const balanceChart = [
    {
      key: 'balanceNative',
      display: `${formatNumber(data.native.value, data.native.exponent)} ${data.native.displayDenom.toUpperCase()}`,
      value: data.native.value,
    },
    {
      key: 'balanceStake',
      display: `${formatNumber(data.stake.value, data.stake.exponent)} ${data.stake.displayDenom.toUpperCase()}`,
      value: data.stake.value,
    },
    {
      key: 'balanceNonce',
      display: `${formatNumber(data.nonce.value, data.nonce.exponent)} ${data.nonce.displayDenom.toUpperCase()}`,
      value: data.nonce.value,
    },
    {
      key: 'balanceVote',
      display: `${formatNumber(data.vote.value, data.vote.exponent)} ${data.vote.displayDenom.toUpperCase()}`,
      value: data.vote.value,
    },
  ];

  return balanceChart;
};
