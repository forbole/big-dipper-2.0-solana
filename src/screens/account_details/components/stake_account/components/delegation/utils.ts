import numeral from 'numeral';
import { DelegationType } from './types';

export const formatBalanceData = (data: DelegationType) => {
  const balanceChart = [
    {
      key: 'balanceActivated',
      display: `${numeral(data.activated.value).format(data.activated.format)} ${data.activated.denom.toUpperCase()}`,
      value: data.activated.value,
    },
    {
      key: 'balanceActivating',
      display: `${numeral(data.activating.value).format(data.activating.format)} ${data.activating.denom.toUpperCase()}`,
      value: data.activating.value,
    },
  ];

  return balanceChart;
};
