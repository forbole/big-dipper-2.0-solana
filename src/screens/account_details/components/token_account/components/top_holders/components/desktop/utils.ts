export const columns:{
  key: string;
  align?: 'left' | 'center' | 'right' | 'justify' | 'inherit';
  width: number;
}[] = [
  {
    key: 'rank',
    width: 5,
    align: 'center',
  },
  {
    key: 'holder',
    width: 65,
  },
  {
    key: 'amount',
    width: 30,
    align: 'right',
  },
];
