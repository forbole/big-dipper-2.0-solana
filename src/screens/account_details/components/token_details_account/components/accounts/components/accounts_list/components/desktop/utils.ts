export const columns:{
  key: string;
  align?: 'left' | 'center' | 'right' | 'justify' | 'inherit';
  width: number;
}[] = [
  {
    key: 'address',
    width: 95,
  },
];
