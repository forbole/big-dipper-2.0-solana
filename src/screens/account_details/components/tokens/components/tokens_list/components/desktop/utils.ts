export const columns:{
  key: string;
  align?: 'left' | 'center' | 'right' | 'justify' | 'inherit';
  width: number;
}[] = [
  {
    key: 'token',
    width: 50,
  },
  {
    key: 'amount',
    align: 'right',
    width: 50,
  },
];
