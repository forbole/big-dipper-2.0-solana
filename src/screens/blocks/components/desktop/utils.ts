export const columns:{
  key: string;
  align?: 'left' | 'center' | 'right' | 'justify' | 'inherit';
  width: number;
}[] = [
  {
    key: 'slot',
    width: 10,
  },
  {
    key: 'leader',
    width: 35,
  },
  {
    key: 'hash',
    width: 30,
  },
  {
    key: 'txs',
    align: 'right',
    width: 10,
  },
  {
    key: 'time',
    align: 'right',
    width: 15,
  },
];
