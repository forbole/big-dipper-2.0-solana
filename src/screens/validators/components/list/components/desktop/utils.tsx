import React from 'react';

export const fetchColumns = (_t): {
  key: string;
  align?: 'left' | 'center' | 'right' | 'justify' | 'inherit';
  width: number;
  component?: React.ReactNode;
  sortKey?: string;
  sort?: boolean;
}[] => {
  return ([
    {
      key: 'idx',
      width: 5,
    },
    {
      key: 'validator',
      sortKey: 'validator.name',
      width: 30,
      sort: true,
    },
    {
      key: 'skipRate',
      sortKey: 'skipRate.rate',
      align: 'right',
      width: 20,
      sort: true,
    },
    {
      key: 'stake',
      sortKey: 'stake',
      align: 'right',
      width: 15,
      sort: true,
    },
    {
      key: 'commission',
      sortKey: 'commission',
      align: 'right',
      width: 15,
      sort: true,
    },
    {
      key: 'lastVote',
      sortKey: 'lastVote',
      align: 'right',
      width: 15,
      sort: true,
    },
  ]);
};
