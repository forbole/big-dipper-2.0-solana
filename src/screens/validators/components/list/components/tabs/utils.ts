export const tabLabels = [
  'active',
  'delinquent',
  'allValidators',
];
