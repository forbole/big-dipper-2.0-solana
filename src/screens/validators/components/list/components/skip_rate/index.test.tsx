import React from 'react';
import renderer from 'react-test-renderer';
import { MockTheme } from '@tests/utils';
import SkipRate from '.';

// ==================================
// mocks
// ==================================

// ==================================
// unit tests
// ==================================
describe('screen: Validators/SkipRate', () => {
  it('matches snapshot', () => {
    const component = renderer.create(
      <MockTheme>
        <SkipRate
          skipRate={{
            rate: 0.4,
            skip: 100,
            total: 200,
          }}
        />
      </MockTheme>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  afterEach(() => {
    jest.clearAllMocks();
  });
});
