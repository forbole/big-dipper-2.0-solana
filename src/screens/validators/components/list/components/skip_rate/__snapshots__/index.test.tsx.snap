// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`screen: Validators/SkipRate matches snapshot 1`] = `
<div
  className="makeStyles-root"
>
  <div
    className="makeStyles-content"
  >
    <p
      className="MuiTypography-root MuiTypography-body1"
    >
      100 / 200
    </p>
    <p
      className="MuiTypography-root percentage MuiTypography-body1"
    >
      40.00
      %
    </p>
  </div>
  <div
    className="makeStyles-chart"
  >
    <div
      className="makeStyles-active"
    />
  </div>
</div>
`;
